api/exchange.go

```diff
@@ -237,6 +237,19 @@ type ExchangeShim interface {
 	FillTrackable
 }
 
+// TradeMetricsHandler is invoked by the MetricsTracker to process new trades
+type TradeMetricsHandler interface {
+	Read(trades []model.Trade)
+	Reset()
+	TotalBaseVolume() float64
+	NumTrades() int
+}
+
+// MetricsTracker knows how to track metrics, including trades
+type MetricsTracker interface {
```

Comment:
I don't think this should be it's own interface either. we take care of handler registrations automatically in `makeFillTracker()` by calling `fillTracker.RegisterHandler(fillLogger)` -- where fillLogger can be our single instance of `tradeMetricsHandler`.